casper/src/main/scala/coop/rchain/casper/EstimatorHelper.scala

```diff
@@ -35,54 +37,88 @@ object EstimatorHelper {
-          uncommonAncestorsMap <- DagOperations.uncommonAncestors[F](blockMetaDataSeq, dag)
-          (b1AncestorsMap, b2AncestorsMap) = uncommonAncestorsMap.partition {
-            case (_, bitSet) => bitSet == BitSet(0)
-          }
-          b1AncestorsMeta    = b1AncestorsMap.keys
-          b2AncestorsMeta    = b2AncestorsMap.keys
-          b1AncestorChannels <- buildBlockAncestorChannels[F](b1AncestorsMeta.toList)
-          b2AncestorChannels <- buildBlockAncestorChannels[F](b2AncestorsMeta.toList)
-          conflicts          = b1AncestorChannels.intersect(b2AncestorChannels).nonEmpty
-          _ <- if (conflicts) {
-                Log[F].info(
-                  s"Block ${PrettyPrinter.buildString(b1.blockHash)} and ${PrettyPrinter
-                    .buildString(b2.blockHash)} conflicts."
-                )
-              } else {
-                Log[F].info(
-                  s"Block ${PrettyPrinter
-                    .buildString(b1.blockHash)}'s channels ${b1AncestorChannels.map(PrettyPrinter.buildString).mkString(",")} and block ${PrettyPrinter
-                    .buildString(b2.blockHash)}'s channels ${b2AncestorChannels.map(PrettyPrinter.buildString).mkString(",")} don't intersect."
-                )
-              }
-        } yield conflicts
+    dag.deriveOrdering(0L).flatMap { implicit ordering =>
+      for {
+        b1MetaDataOpt <- dag.lookup(b1.blockHash)
+        b2MetaDataOpt <- dag.lookup(b2.blockHash)
+        uncommonAncestorsMap <- DagOperations.uncommonAncestors[F](
+                                 Vector(b1MetaDataOpt.get, b2MetaDataOpt.get),
+                                 dag
+                               )
+        (b1AncestorsMap, b2AncestorsMap) = uncommonAncestorsMap.partition {
+          case (_, bitSet) => bitSet == BitSet(0)
+        }
+        b1Events <- extractBlockEvents[F](b1AncestorsMap.keys.toList)
+        b2Events <- extractBlockEvents[F](b2AncestorsMap.keys.toList)
+        conflictsBecauseOfJoins = extractJoinedChannels(b1Events)
+          .intersect(allChannels(b2Events))
+          .nonEmpty || extractJoinedChannels(b2Events).intersect(allChannels(b1Events)).nonEmpty
+        conflicts = conflictsBecauseOfJoins || containConflictingEvents(b1Events, b2Events)
+        _ <- if (conflicts) {
+              Log[F].info(
+                s"Blocks ${PrettyPrinter.buildString(b1.blockHash)} and ${PrettyPrinter
+                  .buildString(b2.blockHash)} conflict."
+              )
+            } else {
+              Log[F].info(
+                s"Blocks ${PrettyPrinter
+                  .buildString(b1.blockHash)} and ${PrettyPrinter
+                  .buildString(b2.blockHash)} don't conflict."
+              )
+            }
+      } yield conflicts
     }
 
-  private def isVolatile(comm: COMM, consumes: Seq[Consume], produces: Seq[Produce]) =
+  private[this] def containConflictingEvents(
```

Comment:
typo?

Reply:
Can't see it

Reply:
`containConflictingEvents` -> `containsConflictingEvents` ?

Reply:
"contain" is correct, because multiple block events are being compared